lib/SPIRV/libSPIRV/SPIRVFunction.cpp

```diff
@@ -149,8 +149,9 @@ SPIRVFunction::decodeBB(SPIRVDecoder &Decoder) {
     }
 
     SPIRVInstruction *Inst = static_cast<SPIRVInstruction *>(Decoder.getEntry());
-    assert(Inst);
-    BB->addInstruction(Inst);
+    assert(Inst);
+    if (Inst->getOpCode() != OpUndef)
+      BB->addInstruction(Inst);

```

Comment:
Why did you think that undef constant will not be added to the module? 
As I see in the implementation of method getEntry(), it is called at this line: M.add(Entry);.